File: frontend/src/app/models/machine-info.ts
import {MessageType} from '../enums/message-type.enum';

export class MachineInfo {
  public id: string;
  public machineName: string;
  public networks: NetworkInfo[];
  public antivirus: SecuritySoftwareInfo[];
  public firewalls: SecuritySoftwareInfo[];
  public windowsVersion: string;
  public netFrameworkVersion: string;
  public storageDevices: StorageDeviceInfo[];
}

export class SecuritySoftwareInfo {
  public name: string;
  public enabled: boolean;
}

export class StorageDeviceInfo {
  public name: string;
  public totalSize: number;
  public availableSize: number;
  public label: string;
}

export class NetworkInfo {
  public adapterName: string;
  public iPv4: string;
}

export class CommandOutput {
  constructor(data: CommandOutput) {
    this.output = data.output;
    this.messageType = data.messageType;
    this.machineId = data.machineId;
  }
  readonly output: string;
  readonly messageType: MessageType;
  readonly machineId: string;
}

File: backend/src/RemoteCLI.Common/Models/MachineInfo.cs
using System.Collections.Generic;
using System.IO;

namespace RemoteCLI.Common.Models
{
    public enum MessageType
    {
        Input,
        Output,
        Error
    }

    public class MachineInfo
    {
        public IEnumerable<SecuritySoftwareInfo> Antivirus { get; set; }
        public IEnumerable<SecuritySoftwareInfo> Firewalls { get; set; }
        public string Id { get; set; }
        public string MachineName { get; set; }
        public string NetFrameworkVersion { get; set; }
        public IEnumerable<NetworkInfo> Networks { get; set; }
        public IEnumerable<StorageDeviceInfo> StorageDevices { get; set; }
        public string WindowsVersion { get; set; }
    }

    public class NetworkInfo
    {
        public string AdapterName { get; set; }
        public string IPv4 { get; set; }
    }

    public class SecuritySoftwareInfo
    {
        public bool Enabled { get; set; }
        public string Name { get; set; }
    }

    public class StorageDeviceInfo
    {
        public long AvailableSize { get; set; }
        public DriveType DriveType { get; set; }
        public string Label { get; set; }
        public string Name { get; set; }
        public long TotalSize { get; set; }
    }
}